### PR TITLE
Manipulate Model._filterKeys using in-place set operations

### PR DESCRIPTION
**Pull request by brianhelba**
Using in-place set operations prevents object references from breaking and is slightly more more efficient.

### CODE REVIEW

File: girder/models/model_base.py [hunk 1]
```diff
@@ -134,15 +134,15 @@ def filter(self, doc, user=None, additionalKeys=None):
         if doc is None:
             return None
 
-        keys = self._filterKeys[AccessType.READ]
+        keys = tuple(self._filterKeys[AccessType.READ])
```
Comment (zachmullen) [inline girder/models/model_base.py]: `tuple` is an immutable type, I think this will break...

Comment (brianhelba) [inline girder/models/model_base.py]: @zachmullen Indeed, this should have been the set constructor. I'll fix.

**Comment by brianhelba** [comment]
Fixed problem and rebased.

**Comment by zachmullen** [comment]
LGTM

**Pull request merged**